### PR TITLE
Add support for Managed, Virtual, Key and null checks

### PR DESCRIPTION
**Pull request by d-sooter**
I found the method in the new CDS version that takes care of the following 
 * - remove virtual fields
 * - add managed, if not provided
 * - assert not nullable without default (for better error message)
 * - add default values, if not provided (mainly for rest response body, odata does read after write)
 * - add UUIDs
 * - propagate keys

I also noticed that the connection and the deployments for multi schemas worked but the read write had an issue. 
i changed the entry point to the Connection generation and it now works fine

### CODE REVIEW

File: lib/pg/Service.js [hunk 1]
```diff
@@ -164,6 +161,10 @@ module.exports = class PostgresDatabase extends cds.DatabaseService {
   async acquire(arg) {
     // const tenant = (typeof arg === 'string' ? arg : arg.user.tenant) || 'anonymous'
     const dbc = await this._pool.connect()
+    // SET SCHEMA
+    if (this.options.credentials && this.options.credentials.schema) {
```
Comment (vobu) [inline lib/pg/Service.js:165]: what do you think about adding tests that run against a schema other than `public` specifically?

e.g., insert another set of `credentials` into https://github.com/sapmentors/cds-pg/blob/ad04e254597a89323e8d3f481cb57c8aa085630d/__tests__/lib/pg/service.test.js#L22  with a dedicated schema on the dockerized db?

_alternatively_, probably code CRUD series à la

0. connect to db with schema `x`
1. insert data into db:`x`
2. update data in db:`x`
3. read data from db:`x`
4. delete data from db:`x`

either way, we'd make sure that we don't break functionality when running against a schema other than `public`.

**Review by vobu — COMMENTED** [review]
time to add tests :)

File: lib/pg/Service.js (hunk 1 above)
Comment (d-sooter) [reply lib/pg/Service.js:165]: Yes we need to, I didnt get to it today but i might find some time on the weekend to add some test.

Comment (vobu) [reply lib/pg/Service.js:165]: want me to let the PR stand until you get to the tests?

Comment (d-sooter) [reply lib/pg/Service.js:165]: Or i can create a new one. I thing the virtual, key, and managed stuff is quite useful

Comment (vobu) [reply lib/pg/Service.js:165]: and all of that sits in `this.before(['CREATE', 'UPDATE'], '*', this._input)`?

Comment (d-sooter) [reply lib/pg/Service.js:165]: yes sap moved it a couple of releases ago it can be found in 
cds-runtime>lib>db>generic>input

Comment (vobu) [reply lib/pg/Service.js:165]: ok, then i'll merge this (also to test the -hopefully fixed- cd gh action 😆 ), create an issue and pin it to you for creating tests.
ok with this?

Comment (d-sooter) [reply lib/pg/Service.js:165]: Sounds good

**Review by vobu — APPROVED** [review]

**Pull request merged**